src/time_series_metadata_store.jl

```diff
@@ -1040,6 +1040,7 @@ function _try_time_series_metadata_by_full_params(
 end
 
 function compare_values(
+    match_fn::Union{Function, Nothing},
```

Comment:
Wait, what is supposed to happen here is a user passes a nothing?

Reply:
We can't just do
```julia
compare_values(x, y; kwargs...) = compare_values(isequivalent, x, y; kwargs...)
```
because we need to be able to keep track of the fact that the two-argument version was initially called so that an old two-argument version in PSY can be called (this was the backwards compatibility issue with my first attempt). So we do
```julia
compare_values(x, y; kwargs...) = compare_values(nothing, x, y; kwargs...)
```
and define that `nothing` will eventually default to `isequivalent` if it isn't intercepted by a two-argument version. This is documented here:
https://github.com/NREL-Sienna/InfrastructureSystems.jl/blob/295e3656f948c80f6d3bcfb10e9b1788e1553c0c/src/utils/utils.jl#L136